tv_apps/tv_shared/components/smart-dialog/script.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+window.SmartDialog = (function(win) {
+
+  // Extend from the HTMLElement prototype
+  var proto = Object.create(HTMLElement.prototype);
+
+  // Bezier curve function
+  var Bezier = function(bezierParam) {
+    var a1 = 3.0 * bezierParam[0];
+    var b1 = 3.0 * (bezierParam[2] - bezierParam[0]) - a1;
+    var c1 = 1.0 - a1 - b1;
+    var a2 = 3.0 * bezierParam[1];
+    var b2 = 3.0 * (bezierParam[3] - bezierParam[1]) - a2;
+    var c2 = 1.0 - a2 - b2;
+    var curveX = function(t) {
+      return ((c1 * t + b1) * t + a1) * t;
+    };
+     
+    var curveY = function(t) {
+      return ((c2 * t + b2) * t + a2) * t;
+    };
+     
+    var calculateCurve = function(x, epsilon) {
+      var t0;
+      var t1;
+      var t2;
+      var x2;
+       
+      t0 = 0.0;
+      t1 = 1.0;
+      t2 = x;
+       
+      if (t2 < t0) {
+        return t0;
+      }
+      if (t2 > t1) {
+        return t1;
+      }
+       
+      while (t0 < t1) {
+        x2 = curveX(t2);
+        if (Math.abs(x2 - x) < epsilon) {
+          return t2;
+        }
+        if (x > x2) {
+          t0 = t2;
+        } else {
+          t1 = t2;
+        }
+        t2 = (t1 - t0) * 0.5 + t0;
+      }
+       
+      return t2;
+    };
+     
+    return function(x) {
+      return curveY(calculateCurve(x, 0.001));
+    };
+  };
+
+  var semiCircleAnimation = function(element, duration, bezierParam, callback) {
+
+    var winWidth = window.innerWidth;
+    var winHeight = window.innerHeight;
+    var finalSize = Math.sqrt((winWidth/2*winWidth/2 + winHeight*winHeight));
+
+    // create svg element
+    var svgEle = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    var existingSvg = document.getElementById(element.id + '-svg');
+    if(existingSvg) {
```

Comment:
nit: add space between `if` and `(`

You may try run `gjslint` for checking these styling.